cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,326 @@
+
+
+### Architecture Diagram
+
+
+### Detailed Design
+
+#### Proposed Option
+
+Radius should support AWS IRSA.
+
+#### Prerequisite
+
+1. [Setup OIDC provider] (https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html) should be complete for the cluster. 
+2. `rad-role` with desired policies should be created. Its trust relation should look similar to below. This establishes a trust relation between the annotated service-accounts in the specified namespace in specified cluster with the specified OIDC provider to be established. 
+  ```
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::817312594854:oidc-provider/oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866:sub": "system:serviceaccount:radius-system:ucp",
+                    "oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866:aud": "sts.amazonaws.com"
+                }
+            }
+        },
+    ]
+}
+  ```
+
+  Note: 67DDAC18D8C44CEDCF1C9719A8E9B866 is the specific eks cluster instance.
+  The aud is the intended recipient of the token.
+  The sub is the entity that the token represents.
+
+
+#### Radius support
+
+**installation**
+
+User can enable IRSA for AWS by by choosing "IRSA" option by choosing IRSA during rad init and providing a RoleARN (`arn:aws:iam::123456789012:role/rad-role`). This stores the roleARN as credential. It also mounts the service-account token to ucp and rp pods as secret.
+
+
+```
+    Containers:
+      ucp:
+        Container ID:   docker://faa5af929f91e053b04b023e05c4fa6363e87688df265443d9f377258fc3fd04
+        Image:          ghcr.io/radius-project/ucpd:latest
+        Image ID:       docker-pullable://ghcr.io/radius-project/ucpd@sha256:991ed703a3260a19535690242cef4d7e4d076038b016443d9106f2d76d57b0da
+        Ports:          9443/TCP, 9090/TCP
+        Host Ports:     0/TCP, 0/TCP
+        :
+        Environment:
+          UCP_CONFIG:                   /etc/config/ucp-config.yaml
+          BASE_PATH:                    /apis/api.ucp.dev/v1alpha3
+          TLS_CERT_DIR:                 /var/tls/cert
+          PORT:                         9443
+        Mounts:
+          /etc/config from config-volume (rw)
+          /var/run/secrets/eks.amazonaws.com/serviceaccount from aws-iam-token (ro)
+          /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-d22ld (ro)
+          /var/tls/cert from cert (ro)
+```
+
+Note: Since AWS IRSA works by mounting the service-account token to pods, the pod spec will be patched and pods recreated if we choose to use rad register to register the roleARN after radius installation.
```

Comment:
I'm scared of this. For Azure we found a solution that doesn't require restarting Radius. What other options do we have here?

Reply:
Its a document mistake. We dont restart for registering credential, but we have to enable IRSA at the time of install. This means if a user has radius already wants to use AWS IRSA the user will have to reinstall.